snuba/utils/streams/batching.py

```diff
@@ -2,38 +2,43 @@
 import time
 from abc import ABC, abstractmethod
 from dataclasses import dataclass
-from typing import Any, MutableMapping, MutableSequence, Optional, Sequence, Tuple
+from typing import (
+    Any,
+    Generic,
+    MutableMapping,
+    MutableSequence,
+    Optional,
+    Sequence,
+    TypeVar,
+)
 
 from confluent_kafka import (
     OFFSET_BEGINNING,
     OFFSET_END,
     OFFSET_INVALID,
     OFFSET_STORED,
-    Consumer,
     KafkaError,
     KafkaException,
-    Message,
     Producer,
-    TopicPartition,
 )
+from confluent_kafka import Message as ConfluentMessage
 
 from snuba.utils.metrics.backends.abstract import MetricsBackend
+from snuba.utils.streams.kafka import KafkaConsumer, KafkaMessage, TopicPartition
 
 
 logger = logging.getLogger("batching-kafka-consumer")
 
-
-DEFAULT_QUEUED_MAX_MESSAGE_KBYTES = 50000
-DEFAULT_QUEUED_MIN_MESSAGES = 10000
+TMessage = TypeVar("TMessage")
 
 
-class AbstractBatchWorker(ABC):
+class AbstractBatchWorker(ABC, Generic[TMessage]):
     """The `BatchingKafkaConsumer` requires an instance of this class to
     handle user provided work such as processing raw messages and flushing
     processed batches to a custom backend."""
 
     @abstractmethod
-    def process_message(self, message: Message) -> Optional[Any]:
+    def process_message(self, message: TMessage) -> Optional[Any]:
```

Comment:
This `Any` will eventually become a generic (@fpacifici suggested `TProcessed` here) but it's not needed yet, so I'm avoiding adding even _more_ complexity to this change — this will happen eventually ,though.

Reply:
I can do that, but I am also very likely going to be updating this in a later change anyway. I'm not sure whether I'm more likely to forget to close the ticket or forget to make the change. 🙂